### ISSUE TITLE
createDocumentSubsegment fails finding document.constructor.modelName when saving an EmbedDocument with no model.

### ISSUE THREAD

**Issue by elribonazo** [unbill/mongoose-xray#2]
Hi, 

When I try to save a subDocument on a Mongoose Model, that EmbedDocument does not contain modelName.

This is my workarround, u may want to include it :)
```javascript
/**
 * Creates the subsegment for query middleware
 * @param operation {String} The operation that is to occur (findOne, update, etc...)
 * @param document {Object} The document related to the operation
 * @param [options] {MongooseXRayOptions} Options for Mongoose-XRay
 */
exports.createDocumentSubsegment = (operation, document, options) => {
  const parent = AWSXRay.getSegment();
  if (parent) {

    const modelName = document.constructor.modelName || `${document.constructor.name}-${document.constructor.path}`;
    const subsegment = parent.addNewSubsegment(
      `${modelName}-${operation}`
    );
    subsegment.addAnnotation('model', modelName);
    subsegment.addMetadata('operation', operation);
    if (options && options.verbose) {
      subsegment.addMetadata('document', JSON.stringify(document));
    }
    logDebugSafe(
      `Mongoose-XRay: Opened Subsegment: ${subsegment.id} Parent Segment: ${parent.id}`
    );
    return subsegment;
  }
};
```

**Comment by eric-swann-q2** [comment]
Awesome thanks @elribonazo! I will add this in.  Also feel free to PR if you find things like this...happy to pull them in.

**Comment by eric-swann-q2** [comment]
@elribonazo This should be fixed in version 3.0.3

I also added in "getQuery" if "getFilter" is undefined for the query middleware.

**Comment by eric-swann-q2** [comment]
Also, help out this project by giving it a star if you wouldn't mind.
